src/App.tsx

```diff
@@ -1,12 +1,25 @@
 import React from "react";
 import { MainPage } from "pages/main";
 import { ErrorBoundary, Footer } from "shared/components";
+import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
+import { ErrorPage } from "pages/error";
+import { RouterPaths } from "shared/enums";
 
 const App: React.FC = () => {
   return (
     <React.StrictMode>
       <ErrorBoundary>
-        <MainPage />
+        <Router>
+          <Switch>
+            <Route path={RouterPaths.HOME} exact component={MainPage} />
+            <Route path={RouterPaths.FILM} component={MainPage} />
+            <Route path={RouterPaths.SEARCH} component={MainPage} />
```

Comment:
i think that the bottom part, where the movie list is rendered, should be outside Switch. Think how to move logic and components to handle this.  But do it in scope of router HT